protocols/v2/framing-sv2/src/lib.rs

```diff
@@ -12,12 +19,38 @@
 //! | `msg_length` | `U24` | Length of the protocol message, not including this header. |
 //! | `payload` | `BYTES` | Message-specific payload of length `msg_length`. If the MSB in `extension_type` (the `channel_msg` bit) is set the first four bytes are defined as a `U32` `"channel_id"`, though this definition is repeated in the message definitions below and these 4 bytes are included in `msg_length`. |
 //!
-//! # Features
+//!
+//! ## Usage
+//!
+//! Nearly all messages sent between Sv2 roles are serialized with the [`framing::Sv2Frame`]. The
+//! exception is when two Sv2 roles exchange Noise protocol handshake messages.
+//!
+//! Before Sv2 roles can communicate securely, they must perform a Noise handshake (note that Noise
+//! encryption is optional for communication between two local Sv2 roles (i.e. a local mining
+//! device and a local mining proxy), but required between two remote Sv2 roles (i.e. a local
```

Comment:
there's nothing really imposing encryption, even if the roles are remote.

```suggestion
//! device and a local mining proxy), but recommended between two remote Sv2 roles (i.e. a local
```

Reply:
I learned the concept of this being required from the [4 Protocol Security](https://github.com/stratum-mining/sv2-spec/blob/main/04-Protocol-Security.md#4-protocol-security) spec in the first section, third paragraph that says:

> Using the handshake protocol to establish secured communication is **optional** on the local network (e.g. local mining devices talking to a local mining proxy). However, it is **mandatory** for remote access to the upstream nodes, whether they be pool mining services, job declarating services or template distributors.

Do I misunderstand this?

Reply:
well if it's in the spec, I take it back